flycheck.el

```diff
@@ -1149,6 +1147,20 @@ Return the path of the file."
         tempfile)
     (flycheck-temp-file-system filename)))
 
+(defun flycheck-temp-files-writable-p (checker)
+  "Whether CHECKER can write temporary files.
+
+If CHECKER has `source' or `source-inplace' in its `:command',
+return whether flycheck has the permissions to create the
+respective temporary files.
+
+Return t if CHECKER does not use temporary files."
+  (let ((args (flycheck-checker-arguments checker)))
+    (and (or (not (memq 'source args))
+             (file-writable-p (flycheck-temp-file-system buffer-file-name)))
+         (or (not (memq 'source-inplace args))
+             (file-writable-p (flycheck-temp-file-inplace buffer-file-name))))))
```

Comment:
Would this read better like this?

```elisp
(cond
  ((memq 'source args) (file-writable-p …))
  ((memq 'source-inplace args) (file-writable-p …))
  (t t))
```

Reply:
I agree it's not optimal, but some checkers have both `source` and `source-inplace` in their arguments.

Reply:
Oh? Hmm. Do we actually have any like that currently? (I couldn't find any) When is that useful?

Reply:
Ah you're right, we don't.  I was thinking of #1354 which uses `source-original` and `source`.  Combining `source` and `source-inplace` does not make much sense after all.